src/login/components/DataPanel/DataPanel.js

```diff
@@ -67,7 +85,10 @@ class DataPanel extends Component {
           margin: "1rem 0 ",
         }}
       >
-        <RenderViewData editDataMode={this.state.editDataMode} />
+        <RenderViewData 
```

Comment:
would it be possible to combine the RenderViewdata/RenderEditData into one?

```
<RenderData
        toggleEditMode={this.toggleEditMode}
         editDataMode={this.state.editDataMode}
 /> 
```

```
const RenderData = (props) => {
  return (
    <Fragment>{props.editDataMode ? <EditDataBox {...props} /> : <ViewDataTable {...props} />}</Fragment>
  );
};

```
it's only my suggestion to reduce code, you might be have some reason for separating functions.

Reply:
yes, that's a great idea! 
But you are right, I am keeping the functions separate while the other elements and components are still in development. However, if these don't need more work, combining them is an excellent suggestion.

Reply:
perfect!